### DISCUSSION TITLE
Edge Caching Issues

### DISCUSSION THREAD

**Discussion by Bobface** [vercel/community#3980] [Help]
### Summary

Hello everybody,

I am having trouble with the caching behaviour of Route Handlers on Vercel using NextJS 13. I am setting

```
Cache-Control: public, max-age=0, s-maxage=1
```
 
on my responses, but this does not seem to be handled in any way by the edge cache. For some routes, I constantly get X-Vercel-Cache: HIT even after waiting for minutes. With others, I only get X-Vercel-Cache: BYPASS. 

The API routes I am talking about here are basic GET routes with no cookies, custom request headers, or anything else. Any idea what could cause this?

### Steps to Reproduce

1. Create a basic GET API route
2. Return the headers as shown above
3. Check the response headers for `X-Vercel-Cache` to see the cache status.

**Comment by amyegan** [comment]
Hey @Bobface. Try using the [`CDN-Cache-Control`](https://vercel.com/docs/edge-network/caching#cdn-cache-control) or `Vercel-CDN-Cache-Control` header. That will let you control the Vercel Edge Cache.

**Reply by Bobface** [reply]
Hey, thanks for your response! I've tried these too, without success. I likely should have mentioned that I am using NextJS 13, so these are actually "Route Handlers".

**Reply by amyegan** [reply]
The default value, `cache-control: public, max-age=0, must-revalidate`, instructs both the Edge Network and the browser not to cache. Is that what you're trying to accomplish? I want to be sure I understand the goal. 

If you have a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that you can share, that would help me get a better view of what's happening.

**Reply by moorage** [reply]
@amyegan I too am getting cache hits on my app/api endpoints for my nextjs apps, despite *not* having any cache headers.  

Can you confirm that the intended behavior is that an app/api endpoint does not cache unless the developer specifies headers to cache?

also, this seems to be related to https://github.com/orgs/vercel/discussions/3259 and https://github.com/orgs/vercel/discussions/4087

<img width="428" alt="Screenshot 2023-09-15 at 12 06 26 AM" src="https://github.com/vercel/community/assets/8590/bc78dc53-a26d-4ef7-8398-92c266ba6519">
